yaml2rst/rst_writer.py

```diff
@@ -0,0 +1,308 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE-SCHEMAS.
+# Copyright (C) 2017 CERN.
+#
+# INSPIRE-SCHEMAS is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE-SCHEMAS is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE-SCHEMAS; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""
+This module defines methods and rules used to parse a ``TreeNode`` to restructured-text strings.
+"""
+
+from __future__ import absolute_import
+import os
+from yaml2rst.rst_util import *
+
+
+PROPERTIES = bold('Properties:')
+ANY_OF = 'May satisfy *any* of the following definitions:'
+ALL_OF = 'Must satisfy *all* of the following definitions:'
+ONE_OF = 'Must satisfy *exactly one* of the following definitions:'
+ENUM = bold('Allowed values:')
+ITEMS = "Elements of {} must satisfy the following definitions:"
```

Comment:
Maybe rather `"Every element of {} is:"` otherwise it looks a bit funny when there is a title and a description.